### PR TITLE
fix: shortened URL for referral

### PR DESCRIPTION
**Pull request by rebelchris**
Add shortened URL for sendgrid.

WT-1980

@idoshamun We only need to worry about the data loader in graphQL requests? (thus can omit like in this PR?)

### CODE REVIEW

File: src/common/mailing.ts [hunk 1]
```diff
@@ -97,12 +97,20 @@ interface EmailContact {
...
+  log: FastifyBaseLogger,
+) => {
+  const rawInviteURL = getInviteLink({
     referralOrigin: 'generic',
     userId: profile.id,
   });
 
+  const [genericInviteURL] = await Promise.all([
+    getShortUrl(rawInviteURL.toString(), log),
+  ]);
```
Comment (capJavert) [inline src/common/mailing.ts:112]: Why promise.all here?

**Review by capJavert — COMMENTED** [review]
Nice, no need for data loader here since you are not requesting same short url multiple times in worker context. Also I don't think we can use it since we only added support in request context for now.

File: src/common/mailing.ts (hunk 1 above)
Comment (rebelchris) [reply src/common/mailing.ts:112]: Ah I actually had 2 items there before, let me adjust it :)

**Review by capJavert — APPROVED** [review]

**Review by idoshamun — APPROVED** [review]
Looks good!

**Pull request merged**